metricshub-web/react/src/components/config/Tree/BackupSetNode.jsx

```diff
@@ -0,0 +1,195 @@
+import * as React from "react";
+import { TreeItem } from "@mui/x-tree-view";
+import { Box, IconButton, Menu, MenuItem } from "@mui/material";
+import MoreVertIcon from "@mui/icons-material/MoreVert";
+import RestoreIcon from "@mui/icons-material/Restore";
+import DeleteIcon from "@mui/icons-material/Delete";
+import FileTypeIcon from "./icons/FileTypeIcons";
+import FileTreeItem from "./FileTreeItem";
+import QuestionDialog from "../../common/QuestionDialog";
+import { useAppDispatch } from "../../../hooks/store";
+import {
+	restoreConfigFromBackup,
+	deleteBackupFile,
+	fetchConfigList,
+} from "../../../store/thunks/configThunks";
+
+/**
+ * Renders one backup set folder (e.g. "backup-20251016-153012") with its kebab menu (Restore all / Delete backup)
+ * and its file children.
+ *
+ * @param {{
+ *   id: string, // timestamp id e.g. "20251016-153012"
+ *   files: Array<{name:string,size:number,lastModificationTime:string}> & {displayName:string}[],
+ *   dirtyByName: Record<string, boolean>,
+ *   filesByName: Record<string, {validation?:any}>,
+ *   onSelect: (name:string)=>void,
+ *   onRename?: (oldName:string,newName:string)=>void,
+ *   onDelete?: (name:string)=>void,
+ * }} props
+ */
+export default function BackupSetNode({
+	id,
+	files,
+	dirtyByName,
+	filesByName,
+	onSelect,
+	onRename,
+	onDelete,
+}) {
+	const dispatch = useAppDispatch();
+
+	// menu + dialogs (keep the set id stable while dialogs are open)
+	const [menuAnchor, setMenuAnchor] = React.useState(null);
+	const [restoreAllOpen, setRestoreAllOpen] = React.useState(false);
+	const [deleteSetOpen, setDeleteSetOpen] = React.useState(false);
+
+	const openMenu = (e) => {
```

Comment:
These handlers are recreated on every render, the component may re render due to Redux or parent updates, it is better to wrap them with useCallback to avoid unnecessary re-renders.